custom_components/google_home/api.py

```diff
@@ -227,3 +229,94 @@ async def update_google_devices_information(self) -> List[GoogleHomeDevice]:
             ]
         )
         return coordinator_data
+
+    async def delete_alarm_or_timer(
+        self, device: GoogleHomeDevice, item_to_delete: str
+    ) -> None:
+        """Deletes a timer or alarm.
+        Can also delete multiple if a list is provided (Not implemented yet)."""
+
+        if device.ip_address is None:
+            _LOGGER.error("Device %s doesn't have IP address!", device.name)
+            return
+
+        if device.auth_token is None:
+            _LOGGER.error("Device %s doesn't have a auth token!", device.name)
+            return
+
+        url = self.create_url(device.ip_address, PORT, API_ENDPOINT_DELETE)
+
+        # We need to remove charset=UTF-8 or else it will return a 400 Bad Request.
+        # I think this is because of the character "/" in the id string.
+        HEADERS.update(
+            {
+                HEADER_CAST_LOCAL_AUTH: device.auth_token,
+                HEADER_CONTENT_TYPE: "application/json",
+            }
+        )
+
+        data = {"ids": [item_to_delete]}
+
+        item_type = item_to_delete.split("/")[0]
+
+        _LOGGER.debug(
+            "Deleting %s from Google Home device %s - %s - " "Raw data: %s",
+            item_type,
+            device.name,
+            url,
+            data,
+        )
+
+        try:
+            async with self._session.post(
+                url, json=data, headers=HEADERS, timeout=TIMEOUT
+            ) as response:
+                if response.status == HTTP_OK:
+                    resp = await response.json()
+                    if resp:
+                        if "success" in resp:
+                            if resp["success"]:
+                                _LOGGER.debug(
+                                    "Successfully deleted %s for %s",
+                                    item_type,
+                                    device.name,
+                                )
+                            else:
+                                _LOGGER.debug(
```

Comment:
Is it the message which will be printed if alarm id doesn't exist? If so, should be a warning as well.

Reply:
It just prints the response message if the we don't get `success` = `true` back. And yes it should probably be a warning so the user knows if it goes wrong. I don't know if explicitly is about if it does not exists.

Reply:
Can you try to run this service with non existing id?

Reply:
It returns `succes` = `true` even it does not exist.

Reply:
So message on 285 should be an error because this shouldn't happen if everything is alright.